File: Server/src/main/java/org/example/Server/Entity/Ad.java
package org.example.Server.Entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "ads")
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Ad {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;
    private String title;
    private Integer file;
    private Integer price;
    private String description;
    private String date;
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "category_id")
    private Category category;
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "own_user_id")
    private User ownUser;
}

File: BoardServer/src/main/kotlin/org/example/boardserver/entity/Ad.kt
package org.example.boardserver.entity

import jakarta.persistence.Entity
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table

@Entity
@Table(name = "ads")
class Ad(

    var title: String,
    var price: Int,
    var description: String? = null,
    var is_favorite: Boolean = false,

    @ManyToOne
    @JoinColumn(name = "department_id")
    val category: Category
) : BaseAuditEntity<Long>()